pkg/restapi/rp/operation/operations.go

```diff
@@ -1296,8 +1309,44 @@ func (o *Operation) listenForConnectionCompleteEvents() { // nolint: gocyclo,cyc
 	}
 }
 
+func (o *Operation) updateCRContext(invitationID string, record *connection.Record) error {
+	crCtx, err := newTransientStorage(o.transientStore).GetConsentRequest(invitationID)
+	if err != nil {
+		return fmt.Errorf("unable to fetch consentRquestCtx data transient storage: %w", err)
+	}
+
+	crCtx.RPPairwiseDID = record.MyDID
+	crCtx.UserDID = record.TheirDID
+	crCtx.ConnectionID = record.ConnectionID
+
+	err = newTransientStorage(o.transientStore).Put(crCtx.InvitationID, crCtx)
+	if err != nil {
+		logger.Errorf("failed to update invitation data in transient storage : %s", err)
+	}
+
+	if crCtx.SupportsWACI {
+		// TODO remove this once AFG implements https://github.com/hyperledger/aries-framework-go/issues/2860
+		err = o.persistenceStore.Put(getConnToCtxMappingDBKey(record.ConnectionID),
+			[]byte(crCtx.InvitationID))
+		if err != nil {
+			logger.Errorf("failed to update connectionID to ctx id : %s", err)
+		}
+	}
+
+	// consent request doesn't always apply, ex: remote wallet app connection through deep link invitation.
+	if crCtx.CR != nil {
+		err = o.persistenceStore.Put(getConnToTenantMappingDBKey(record.ConnectionID),
+			[]byte(crCtx.CR.Payload.Client.ClientID))
+		if err != nil {
+			logger.Errorf("failed to update connectionID to rp client id : %s", err)
+		}
+	}
+
+	return nil
+}
+
 //nolint:funlen,cyclop
-func (o *Operation) presentProofListener(ppActions chan service.DIDCommAction) {
+func (o *Operation) presentProofListener(ppActions chan service.DIDCommAction) { // nolint:gocyclo
```

Comment:
aren't you updating how webredirect info has been sent ? did it work right away?

Reply:
I did, but it looked like there's some more refactoring to do to clean up the crCtx etc, there are old TODOs, but I didn't want to make potentially-breaking changes in this PR